### PR TITLE
Add auth service and its config

### PR DESCRIPTION
**Pull request by harikt**
This was inspired from @brandonsavage work.

### CODE REVIEW

**Comment by harikt** [comment]
May be will close https://github.com/auraphp/Aura.Auth/pull/58 ?

**Comment by brandonsavage** [comment]
This is a good start. I have a few thoughts on it that I'll be sharing today. Note that none of my comments are meant to be rude!

**Comment by harikt** [comment]
@brandonsavage feel free to comment :) .

We are trying to make things better ;) . Looking forward to hear you.

File: src/AuthService.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+use Aura\Auth\Adapter\AdapterInterface;
+use Aura\Auth\Auth;
+use Aura\Auth\Service\LoginService;
+use Aura\Auth\Service\LogoutService;
+use Aura\Auth\Service\ResumeService;
+use Aura\Auth\Status;
+
+class AuthService
+{
+    private $auth;
```
Comment (brandonsavage) [inline src/AuthService.php]: Private denotes an internal property or method that shouldn't be used by children. This keeps me from extending the class and using these methods. Consider protected.

File: src/AuthService.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    private $login_service;
+
+    private $logout_service;
+
+    private $resume_service;
+
+    private $resumed = false;
+
+    private $adapter;
```
Comment (brandonsavage) [inline src/AuthService.php]: I don't see this property used anywhere in the class.

File: src/AuthService.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    public function __construct(Auth $auth, LoginService $login_service, LogoutService $logout_service, ResumeService $resume_service, AdapterInterface $adapter)
```
Comment (brandonsavage) [inline src/AuthService.php]: Consider breaking this into multiple lines to aid readability.

Comment (harikt) [inline src/AuthService.php]: Sure. I hope you noticed the property in the construct .

File: src/AuthService.php (hunk 1 above)
Comment (harikt) [inline src/AuthService.php]: Yup sure. Noted.

File: src/AuthService.php [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    {
+        return $this->logout_service->forceLogout($this->auth, $status);
+    }
+
+    /**
+     *
+     * Magic call to all auth related methods
+     *
+     */
+    public function __call($method, array $params)
```
Comment (brandonsavage) [inline src/AuthService.php:62]: What would be the primary use case for this method?

Comment (harikt) [inline src/AuthService.php:62]: If people don't like to call `$auth_service->getAuth()->isValid()` , they can use `$auth_service->isValid()` , same for other methods inside `Aura\Auth\Auth` . Just for convenience.

File: src/AuthService.php (hunk 3 above)
Comment (brandonsavage) [inline src/AuthService.php]: I see the adapter being passed in, but I don't know that we need it. 

The Law of Demeter says we shouldn't teach this objet things it doesn't need to know, and the adapter is entirely encapsulated inside the services themselves.

Comment (harikt) [inline src/AuthService.php]: aah I get you.

Thank you for pointing the error.

**Comment by harikt** [comment]
@pmjones I know you like to concentrate on one thing at a time. Still I am disturbing you, apologies. Would love if you have some feedback / thoughts if you like to merge.

The reason is I have written the same class over https://leanpub.com/aurav2/read#leanpub-auto-building-service-class and wished if things are merged, it needs less code / configuration.

**Pull request closed without merging**